src/Commands/SalesFileGenerationCommand.php

```diff
@@ -226,4 +229,35 @@ private function generateSalesFiles(array $config, Collection $stores, string $d
             $this->info($file.' has been created');
         });
     }
+
+    protected function validateAndGetSales(Collection $sales, string $date): Collection
+    {
+        $afterCurrentDate = Carbon::parse($date)->startOfDay()->toDateTimeString();
+        $beforeCurrentDate = Carbon::parse($date)->endOfDay()->toDateTimeString();
+        $paymentTypes = implode(',', PaymentType::values());
+        $validator = Validator::make($sales->values()->all(), [
+            '*.happened_at' => ['required', 'date', 'date_format:Y-m-d H:i:s', "after_or_equal:{$afterCurrentDate}", "before_or_equal:{$beforeCurrentDate}"],
+            '*.net_amount' => ['required', 'numeric', 'decimal:0,2'],
+            '*.discount' => ['required', 'decimal:0,2'],
+            '*.SST' => ['required', 'decimal:0,2'],
+            '*.payments' => ['required', "array:{$paymentTypes}"],
+            '*.payments.cash' => ['required', 'decimal:0,2'],
+            '*.payments.tng' => ['required', 'decimal:0,2'],
+            '*.payments.visa' => ['required', 'decimal:0,2'],
+            '*.payments.mastercard' => ['required', 'decimal:0,2'],
+            '*.payments.amex' => ['required', 'decimal:0,2'],
+            '*.payments.voucher' => ['required', 'decimal:0,2'],
+            '*.payments.others' => ['required', 'decimal:0,2'],
```

Comment:
Can we go via enum instead of direct string?

Reply:
Yes. Made it dynamic.